plugins/bnet_plugin/bnet_plugin.cpp

```diff
@@ -508,7 +517,7 @@ namespace eosio {
            if( fc::time_point::now() - s->block->timestamp  < fc::seconds(6) ) {
            //   ilog( "queue notice to peer that we have this block so hopefully they don't send it to us" );
               auto itr = _block_status.find(s->id);
-              if( itr == _block_status.end() || !itr->received_from_peer ) {
+              if( !_remote_request_irreversible_only && ( itr == _block_status.end() || !itr->received_from_peer ) ) {
```

Comment:
is this necessary?  is there a reason we wouldn't send block notices to another connection?  the semantics are "I know of this block, don't send it" and if the peer is getting reversible blocks from elsewhere and you don't offer notice then they may relay them to you even if they only want irreversible blocks from you